### ISSUE TITLE
Global token scale factor

### ISSUE THREAD

**Issue by yorkshirelandscape** [foundryvtt/foundryvtt#11525]
### User Experience

While acknowledging that the whitespace surrounding the new dynamic tokens is a deliberate choice, it is nonetheless a jarring change for many who are accustomed to standard sized tokens occupying a full grid square. Different modules can, of course, implement their own image mappings to adjust their tokens back to the earlier sizing, but then they run the risk of mixing tokens with those from other sources and having size mismatches. The recently release Starfinder Second Edition Playtest Adventure module was released with a macro to size up all tokens on the screen to solve for this. Rather than leave developers having to include hacks like this to achieve their desired aesthetic, would you consider implementing a global dynamic token scale factor? 

- The simplest approach would be to make it a 1 : 1.2 toggle, but perhaps there's some rationale for allowing arbitrary scaling. 
- It wouldn't change the scale factor shown in the token configuration. 
  - A regular token would still show as having a scale of 1.0, but with the global scale factor of 1.2 enabled, the ring would fully occupy the grid square as many are accustomed to seeing it. 
- It would only affect dynamic tokens because, presumably, if people want their static token to occupy a full square, they'd have created it that way. 
  - Either that or you could add an option to the static token configuration screen to ignore the global scale factor. 

Thanks for considering the request and keep up the good work!

**Comment by Mystler** [comment]
I'd like to add to this by saying it would probably be even better to have scaling settings for both dynamic and standard tokens. That way, when mixing dynamic tokens and standard tokens for whatever reason (e.g. custom dynamic PC and NPC tokens and premium module standard tokens), the GM can decide whether to scale dynamic tokens up for the "full square experience" or scale classic tokens down for the new convention.

**Comment by silvative** [comment]
One point for consideration- you cannot presume that all token rings are the same width, in fact the two default rings have different widths. Any attempt to resolve this by scaling would require a different scaling value to be set for each ring asset, so it would need to be incorporated into the rings.json file that sets the default colour paths and so on.

Solutions like using macros to set a 1.2 scale across the board are clumsy and can only work for one specific ring, so they will have severely detrimental effects for users using other rings, including rings provided by other modules. I would not recommend that approach to content developers or end users in any scenario.

**Comment by yorkshirelandscape** [comment]
> One point for consideration- you cannot presume that all token rings are the same width, in fact the two default rings have different widths. Any attempt to resolve this by scaling would require a different scaling value to be set for each ring asset, so it would need to be incorporated into the rings.json file that sets the default colour paths and so on.

Ahh, right you are. So there's the argument for arbitrary scaling. Knew there must be one.

**Comment by racehd** [comment]
I think this is a fairly complex topic to unpack but here's some thoughts. 

Current Implementation Issues:
1. Non-Intuitive
  The current implementation of dynamic rings is not intuitive. Users expect the rings to fill the grid completely. However, with dynamic rings, the ring does not fill the grid unless a significantly large ring is used. This creates additional complexity for GM's and players who expect the ring to fill the grid. 
2. Aesthetic Concerns
  Even if all tokens and rings are the same size, the current dynamic rings are too small, resulting in a less appealing aesthetic. The ring does not provide a visually satisfying border and makes the token appear undersized within the grid.
3. Accessibility Issue
  The current dynamic ring implementation allocates only 34.9% of the grid area to the artwork, while 65.1% is reserved for outer padding. This significantly reduces the visible space for token art, which is particularly problematic in large, open maps where zooming in is not practical for GM's or players. 

Proposed Solution: 
1. Grid-based Scaling:
  Scale the ring's default value (1.0) based on the grid size so that the ring's edge aligns with the grid's border. This ensures that the ring fills the grid as users expect.
2. Inner Mask File:
  Include an inner mask file with each dynamic ring. This file will define the expected (circular) area for the artwork within the ring, allowing for consistent and accurate placement of the subject artwork. 
3. Dynamic Scaling and Composition:
  - Scale the ring based on the grid size.
  - Use the inner mask file to calculate the appropriate area for the artwork.
  - Automatically scale the artwork to fit within the ring, ensuring the maximum visual space for the art while considering the ring's size.

I believe this will result in a simpler end-user experience to match user expectations while allowing for interchanging of dynamic rings of various types and sizes. It will also maximize the visual space of art within the grid which will improve both aesthetics and accessibility.

**Comment by aaclayton** [comment]
I understand this is a contentious issue with varying opinions on all sides. I'm going to tag this to V13 specifically so that we address as a team and make a decision. No guarantees at this point about what outcomes may occur, but we will discuss it further.

**Comment by racehd** [comment]
Thank you Atropos, 
I wanted to expand on my numbers since they were questioned this weekend. I previously reported that the dynamic ring currently only allocates 34.9% of the grid for the subject area. To give fuller context for discussion, I have calculated additional values:

- Dynamic rings have a 34.9% subject area, and a 65.1% padding area (representing ring + blank space). 
- Abomination Vaults has a 55.4% subject area, and 44.6% padding. This is 37% more space allocated to subject area than Dynamic Rings. 
- My personal token I use has a 58.8% subject area and a 41.2% padding. This is 40.6% more space allocated to subject area than Dynamic Rings
- A thick frame like from Paizo's Gatewalkers product has 41.5% subject area, and 58.5% padding. This is 15.8% more space allocated to subject area than Dynamic Rings
- Pf2e Bestiaries also had a thicker frame, with 42.4% subject area, and 57.6% padding. This is 17.6% more space allocated to subject area than Dynamic Rings

To make this more relatable (and fun!) to discuss, consider this in terms of pizza math. Imagine a standard pizza box of 14x14 inches. Now imagine you invite some friends over to play some games and everyone gets hungry so you order a large pizza expecting it to fill the box. The pizza arrives, everyone gathers around, and when you open the box you find a 9.34-inch pizza pie. Your friends might be disappointed wondering why you didn't order a larger pizza. 

It turns out the large pizza you ordered is now a 9.34-inch pizza pie, with the same area as the current dynamic rings implementation for the 2/3 subject area. If pizza companies could dynamically scale their pies to make them look larger, they certainly would to avoid disappointing their customers with a small pizza in a large box. Similarly, by dynamically scaling the subject area within tokens, Foundry VTT could enhance the visual appeal and usability of tokens, avoiding the current issue where a significant portion of the grid is unused padding.

I understand there is already a module to rescale the tokens, but I maintain my above suggestion that scaling the ring to the grid, and dynamically scaling the art accordingly will be a more intuitive, aesthetically pleasing, and more accessible.  My calculations are attached.  
Thanks again for your time. 
[Pizza Math aka Token Area Analysis.xlsx](https://github.com/user-attachments/files/16500364/Pizza.Math.aka.Token.Area.Analysis.xlsx)

**Comment by lholden** [comment]
This is made even weirder right now with the broken interactions between the PF2E Bestiary and the PF2E Monster Core token modules. 

When you have both modules installed, the Core tokens end up super tiny. When using just the Core tokens they are the normal Foundry 2/3 size.

Then the Bestiary tokens themselves have different scaling based on the art and how much they want to show outside of the borders. For example, somewhere between 1.0 and 1.5. The Core tokens then use 2.0 as their scale so they can have more art outside of the tokens while keeping to the 2/3 token size the dynamic rings default to.

Compared to my PC dynamic tokens which are all 1.0 scale atm.

I think you would need to be able to set the default scaling per module or something, even without the buggy interactions between those two token packs.

**Comment by shemetz** [comment]
Some more food for thought -

CzePeku (a popular creator) is releasing a new project - a token pack subscription that allows users to customize their own tokens, a little bit like TokenStamp's tool but designed around modular tokens.  [See this demo for some examples](<https://www.czepeku.com/tokens/demo>).

CzePeku's token frames have varying widths, but they all share a standard **1-to-7 ratio (14%)** between the transparent outer padding and the inner ring art.  Compare this to the newly recommended ratio by Foundry -- **1-to-3 ratio (33%)** -- and to the old "touching the grid" ratio of about **0-1%** (=no padding between the ring and the grid).

![image](https://github.com/user-attachments/assets/80e7c2f3-4edd-439e-abbb-6c48ad7b9555)

Users who want to use these new tokens, and don't take special care to change the scaling of each one, will find that they have smaller rings than the old standard but larger rings than the new standard.  There's too many standards! :D 


Whichever solution we end up with, I think the users should have tools to easily rescale collections of tokens to whichever standard they enjoy, without having to go through them one by one.  Ideally this would be a passive client-side setting, rather than e.g. a one-time macro that multiplies the scale value of every actor's prototype token.  (Plus, this way different players could customize based on individual preferences)

**Comment by aaclayton** [comment]
I have a lot I could say about this topic but I'm pressed for time so my response will have to be brief for now. I think folks are a bit too focused on the 2/3 convention as a signal of what size the tokens are ultimately rendered on the canvas itself. 

The primary purpose for the 2/3 convention - and why it leaves ample space on the canvas outside of the subject circle - is to make it easy to have dramatic overflow or pop-out for token artwork without needing to configure special subject scale overrides frequently. We can, either as a core change or as something a system or module does, change the final rendered size of the token relative to the grid. 

TL;DR the purpose for the 2/3 convention is about *how is the artwork made* not necessarily about *how should it look when placed in a scene*

**Comment by yorkshirelandscape** [comment]
That great to hear, Atropos. The messaging to this point has left people with the understanding that all token packs conforming to the new standard would feature subject art substantially smaller than had previously been the norm and there would be no easy and accurate way to scale them up so that the ring abuts the grid square due to the variability in ring width. The gist seemed to be that it was better that way and people would get used to it. I hope this indicates there will be some flexibility to how the tokens appear on the canvas.

**Comment by xyzzy42** [comment]
When then first came up I thought it would be possible to dynamically scale the tokens so that either the dynamic token ring outer diameter was scaled to 100% of the grid size, or alternately so that the outer diameter of a static token drawn to 100% grid size would match the outer diameter of a dynamic ring.

Many were dismissive of this being possible, but undeterred, I went and did it anyway.  I'm happy with how it turned out.  You can see here a demo of two different widths of token ring.  

There is no UI other than what you see in the demo.  I manually calculate no scale factors nor tweak any settings to the tokens used.  The setting can be changed dynamically, as shown in the demo, and is a per-client setting.  Different users can make different choices based on their aesthetic preferences, screen size, and visual acuity.

[dynamic-scale-demo-steel.webm](https://github.com/user-attachments/assets/df541666-f609-4812-959b-37af63eae726)
[dynamic-scale-demo-runes.webm](https://github.com/user-attachments/assets/076bde3e-9e06-41b5-980c-fb9305bc2174)

**Comment by yorkshirelandscape** [comment]
With the release of Dynamic Ring fit mode, I think we have the solution everyone wanted. Thank you for your patience and willingness to hear our concerns.
